adr/0021-rules-for-entity-mapping-in-service-catalog.md

```diff
@@ -0,0 +1,175 @@
+
+## Context and Problem Statement
+
+Backstage offers multiple resource types which can be used to represent entities and their relations in service catalog. We need to have a set of rules present in Operate First which will provide context and guidance on how entities are mapped to Backstage resources.
+
+## Decision Drivers
+
+* Easy and clear guidelines and distinction for each resource type.
+* Supports various caveats  like ODH meta operators and Cluster as a service/resource.
+
+## Decision Outcome
+
+[Backstage provides](https://backstage.io/docs/features/software-catalog/descriptor-format) numerous resource types, which has strictly defined relations.
+
+```mermaid
+erDiagram
+Domain ||--o{ System : contains
+System ||--o{ Component : contains
+Component |o--o{ Component : hasSubcomponent
+Component |o--o{ Component : dependsOn
+Component ||--o{ Resource : requires
+Component ||--o{ API : provides
+Component }|--|| API : consumes
+Resource ||--o{ Component : dependsOn
+Resource ||--o{ Resource : dependsOn
+Location
+Template
+Group |o--|{ User : hasMember
+Group |o--|{ Group : hasMember
+```
+
+Relations are generated from resource manifests. Extensive documentation on all supported and recognized relations is available [upstream][3].
+
+Additionally:
+
+`Group` and `User` resources own any other resource type. All resources except `Group`, `User` and `Location` is required to have an owner specified.
+
+### Kind: Domain
+
+This entity represents top level non-root entity within the cloud. Think of it as a SIG type of entity. A whole collection of applications or group of services that are meant to work together and be used together.
+
+Examples:
+
+* **SRE Domain**: Contains all monitoring applications, CI/CD etc, Service Catalog, etc.
+* **Data Science Domain**: Contains ODH and, Kubeflow, Data cataloging solutions and other analytics tools.
+* **Thoth Station Domain**: All Thoth related systems including repository storages like Pulp, everything supporting Thoth.
+
+`Domain` owns multiple `System`s.
+
+### Kind: System
+
+System means a collection of applications, components which together builds a singular service. A system is meant as an interaction point for users as an application.
+
+Examples:
+
+* **Service Catalog System**: Represents Backstage as a whole.
+* **ODH System**: Represents ODH as a collection of its components.
+* **ArgoCD System**
+* **Prow System**: Contains all Prow components like Deck, Crier, Horologium, etc.
+
+`System` resource belongs to a one domain. `System` resource owns `Component` resources.
+
+If there are multiple instances of the same service system, either for different maturity level environments (staging/production) or intended to different audiences (ODH in Smaug and ODH in OS Climate) the System or Component resources need to be duplicated.
+
+### Kind: Component
+
+Individual service, a runtime, component. Usually describes an individual deployment. A component should map to
```

Comment:
I don't think the sentence is finished here or are you referring to the examples below?